Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -1952,10 +1952,11 @@ void HandleBreakEventSet (Event[] es, bool dequeuing)
 						}
 						
 						if (hasBreakInfo) {
-							if (currentStepRequest != null &&
-							    currentStepRequest.Depth != StepDepth.Out &&
+							if (currentRequest != null &&
+								(currentRequest is StepEventRequest currentStepRequest1) &&
+								currentStepRequest1.Depth != StepDepth.Out &&
```

Comment:
You don't need the `currentRequest != null` because that will be handled by `(currentRequest is StepEventRequest currentStepRequest1)`

Reply:
Done! :)